### PR TITLE
apps/lib/s_socket.c: Alias getpid with _getpid for _WIN32

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

**Comment by levitte** [comment]
This was found thanks to #15709

File: apps/lib/s_socket.c [hunk 1]
```diff
@@ -26,6 +26,14 @@
 typedef unsigned int u_int;
 #endif
 
+#ifdef _WIN32
+/*
+ * With MSVC, certain POSIX functions have been renamed to have an underscore
```
Comment (slontis) [inline apps/lib/s_socket.c:31]: I assume this is particular versions that need this?

Comment (levitte) [reply apps/lib/s_socket.c:31]: Nope.

**Comment by slontis** [comment]
So it does actually work it is just a deprecated alias.

The Microsoft-implemented POSIX function name getpid is a deprecated alias for the _getpid function. By default, it generates Compiler warning (level 3) C4996. The name is deprecated because it doesn't follow the Standard C rules for implementation-specific names. However, the function is still supported.

We recommend you use _getpid instead. Or, you can continue to use this function name, and disable the warning. For more information, see Turn off the warning and POSIX function names.

**Review by slontis — APPROVED** [review]

**Comment by levitte** [comment]
> So it does actually work it is just a deprecated alias.

We have a number of those, sprinkled a little here and there:

``` console
$ git grep -E '^# *define +([a-z_]+) +_\1'
apps/cmp.c:# define access _access
crypto/bn/bn_exp.c:#  define alloca _alloca
crypto/conf/conf_def.c:#  define stat    _stat
crypto/conf/conf_sap.c:# define strdup _strdup
crypto/cryptlib.c:#  define alloca _alloca
crypto/rand/randfile.c:#  define stat    _stat
crypto/rand/randfile.c:#  define chmod   _chmod
crypto/rand/randfile.c:#  define open    _open
crypto/rand/randfile.c:#  define fdopen  _fdopen
crypto/rand/randfile.c:#  define fstat   _fstat
crypto/rand/randfile.c:#  define fileno  _fileno
crypto/x509/by_dir.c:#  define stat _stat
e_os.h:#   define open _open
e_os.h:#   define fdopen _fdopen
e_os.h:#   define close _close
e_os.h:#    define strdup _strdup
e_os.h:#   define unlink _unlink
e_os.h:#   define fileno _fileno
engines/e_capi.c:#  define alloca _alloca
engines/e_loader_attic.c:# define stat _stat
providers/implementations/storemgmt/file_store.c:# define stat _stat
test/conf_include_test.c:#  define chdir _chdir
test/testutil/driver.c:# define strdup _strdup
```

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Proposing for urgent merge as this fixes AppVeyor build failure.

**Comment by levitte** [comment]
I agree on urgent.  Will merge promptly

**Comment by levitte** [comment]
Merged

814b5133e9aca90f1edb99c38a26e55cd7e50e19 apps/lib/s_socket.c: Alias getpid with _getpid for _WIN32

**Pull request closed without merging**